src/ReactiveDomain.Identity/ReadModels/PrincipleWrapper.cs

```diff
@@ -0,0 +1,29 @@
+﻿using ReactiveDomain.Util;
+using System.DirectoryServices.AccountManagement;
+
+namespace ReactiveDomain.Identity.ReadModels
+{
+
+    public interface IPrinciple
+    {
+        string Provider { get; }
+        string Domain { get; }
+        string SId { get; }
+    }
+    public class PrincipleWrapper : IPrinciple
```

Comment:
It doesn't look like this class is used anywhere. Is it needed?

Reply:
it supports testing the readmodel

Reply:
I see that the `IPrinciple` interface is used for the tests, but `PrincipleWrapper` isn't used anywhere. Was that intentional?

Reply:
more tests are pending.
It is not possible to instantiate the MS principle class without a call to AD for a setup user.
So without both the interface and a concrete class any tests simply are not possible.